src/components/file-input/__internal__/file-upload-status/file-upload-status.component.tsx

```diff
@@ -0,0 +1,134 @@
+  status: "uploading";
+  /** a number from 0-100 giving the current upload progress as a percentage. Only used for the `uploading` status.
+   * If the progress prop is not specified in the `uploading` status, a loading animation will be shown instead
+   * (or text equivalent for users with a reduced-motion operating system preference).
+   */
+  progress?: number;
+}
+
+interface StatusDoneProps extends LinkProps {
+  /** the status of the upload */
+  status: "completed" | "previously";
+  /** the URL opened by the file link. Must be provided for only the `completed` and `previously` statuses. */
+  href: string;
+}
+
+interface StatusErrorProps {
+  /** the status of the upload */
+  status: "error";
+}
+
+interface MandatoryStatusProps {
+  /** the name of the file */
+  filename: string;
+  /** a function to be executed when the user clicks the appropriate action button (Clear/Delete File/Cancel Upload)  */
+  onAction: () => void;
+  /** The status message. Used to display the current upload progress, including error messages where appropriate. Not used for the `previously` status. */
+  message?: string;
+  /** The icon to use for the file during or after upload */
+  iconType?: IconType;
+}
+
+export type FileUploadStatusProps = MandatoryStatusProps &
+  (StatusUploadingProps | StatusErrorProps | StatusDoneProps);
+
+export const FileUploadStatus = ({
+  status,
+  filename,
+  message,
+  onAction,
+  iconType = "file_generic",
+  ...statusProps
+}: FileUploadStatusProps) => {
+  const locale = useLocale();
+  const statusMessage = message || locale.fileInput.fileUploadStatus();
+
+  let buttonText;
+  let linkProps;
+  let progressBar = null;
+  switch (status) {
+    case "uploading":
+      buttonText = locale.fileInput.actions.cancel();
+      progressBar =
+        (statusProps as StatusUploadingProps).progress === undefined ? (
+          <LoaderBar size="small" />
+        ) : (
+          <ProgressTracker
+            size="small"
+            progress={(statusProps as StatusUploadingProps).progress}
+            length="100%"
+          />
+        );
+      break;
+    case "previously":
+    case "completed":
+      buttonText = locale.fileInput.actions.delete();
+      linkProps = { ...statusProps, icon: iconType };
+      break;
+    case "error":
+      buttonText = locale.fileInput.actions.clear();
+      break;
+    // istanbul ignore next
+    default:
+      // no other cases if consumers are using TS, but ESLint still insists on it
+      break;
+  }
+  const actionButton = (
+    <ButtonMinor onClick={onAction} buttonType="tertiary">
+      {buttonText}
+    </ButtonMinor>
+  );
+  const fileLink = linkProps ? (
+    <Link {...linkProps}>{filename}</Link>
+  ) : (
+    <>
+      <Icon type={iconType} />
+      <span>{filename}</span>
+    </>
+  );
+  const mainRow =
+    status !== "previously" ? (
+      <StyledFileUploadStatusRow>
+        <StyledTypography as="p" mb={0} aria-live="polite">
```

Comment:
question: is this something we could just support in general for `Typography` rather than importing the SC here?

Reply:
Let's address this in the future (if we need it at all) as this work is urgent etc